TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datatypes/UniqueRequestKey.java

```diff
@@ -2,13 +2,17 @@
 
 import lombok.AllArgsConstructor;
 
-@AllArgsConstructor
 public class UniqueRequestKey {
-    public final String connectionId;
+    public final TrafficStreamKey trafficStreamKey;
     public final int requestIndex;
```

Comment:
nit: I will be continually having to remind myself what a `requestIndex` and a `trafficStreamIndex` for TrafficStreamKey are in the context of the Replayer, would love a couple short comments

Reply:
In subsequent edits, these become 
```
    public final TrafficStreamKeyWithRequestOffset trafficStreamKeyAndOffset;
    public final int replayerRequestIndex;
```

Does replayerRequestId help more?  I'll think about what more I can add too.

Reply:
I may just not have a good understanding of what that field is. Is replayerRequestIndex the number of the request for the given traffic stream or for the given connection? Or something else maybe?

Reply:
replayerRequestIndex is used to sequence requests onto the netty worker threads that will transform and send the requests.  This must be indexed from 0 so that we know when we've gotten the first record.  In later commits, that's happening and there's a separate field to track what the TrafficStream's request offset was.  That allows this UniqueRequestKey to show the current offset that the process is aware of + the captured request index as it came in from the server.  That's regardless of whether we got the first TrafficStream object or a subsequent one thanks to the trafficStreamIndex.